dlt/common/storages/fsspec_filesystem.py

```diff
@@ -299,20 +313,28 @@ def glob_files(
         if md["type"] != "file":
             continue
 
+        scheme = bucket_url_parsed.scheme
         # make that absolute path on a file://
         if bucket_url_parsed.scheme == "file" and not file.startswith("/"):
             file = f"/{file}"
-        file_name = posixpath.relpath(file, bucket_url_no_schema)
-        file_url = bucket_url_parsed._replace(
-            path=posixpath.join(bucket_url_parsed.path, file_name)
-        ).geturl()
+
+        if is_unc_path:
+            file_url = "file:///" + file
+            file_name = file.replace(bucket_url_no_schema, "").replace("\\", "/").lstrip("/")
```

Comment:
It's the most straightforward and stupid solution, but I think it'll work. For UNC we're getting `file` from `glob`, which is supposed to always return the same result, so `str` edits like this should work. This way it produces the same result as `relpath` for normal paths.

Reply:
what is `file` here? file_url must contain UNC path so we are able to use it to open later. make sure it is like that